src/libraries/System.Net.HttpListener/tests/HttpListenerTests.cs

```diff
@@ -155,5 +156,30 @@ public void EndGetContext_AlreadyCalled_ThrowsInvalidOperationException()
                 Assert.Throws<InvalidOperationException>(() => listener.EndGetContext(beginGetContextResult));
             }
         }
+
+        [Fact]
+        //[OuterLoop]
+        public void GetContext_StopIsCalled_GetContextUnblocked()
+        {
+            using var listenerFactory = new HttpListenerFactory();
+            var listener = listenerFactory.GetListener();
+            listener.Start();
+
+            var listenerTask = Task.Run(() =>
+            {
+                try
+                {
+                    var context = listener.GetContext();
+                }
+                catch (Exception)
+                {
+                }
+            });
+
+            Task.Delay(1000).Wait(); // let listenerTask to call GetContext
```

Comment:
Using .Wait() can lead to infinite hangs in the test if the product/test code has a bug. We usually add a timeout and then test the return value of Wait() to see if the timeout expired. If so, the test fails.

We have examples in the repo with other tests to use a timeout on Wait().

Diagnosing hanging tests is CI is very painful. So, we try to avoid that.

Reply:
I added a timeout to Wait. There is no need to check whether it actually timed out.